### PR TITLE
Unity3dAPI: added receipt search

### PR DESCRIPTION
**Pull request by noescape00**
Pretty straight forward, it just calls receipt search from SC controller

### CODE REVIEW

File: src/Stratis.Features.Unity3dApi/Controllers/Unity3dController.cs [hunk 1]
```diff
@@ -511,6 +516,32 @@ public LocalExecutionResult LocalCallSmartContractTransaction([FromBody] LocalCa
...
+        /// <param name="topics">The topics to search. All specified topics must be present.</param>
+        /// <param name="fromBlock">The block number from which to start searching.</param>
+        /// <param name="toBlock">The block number where searching finishes.</param>
+        /// 
+        /// <returns>A list of receipts for transactions relating to a specific smart contract and a specific event in that smart contract.</returns>
+        [Route("api/[controller]/receipt-search")]
+        [HttpGet]
+        public async Task<List<ReceiptResponse>> ReceiptSearchAPI([FromQuery] string contractAddress, [FromQuery] string eventName, [FromQuery] List<string> topics = null, [FromQuery] int fromBlock = 0, [FromQuery] int? toBlock = null)
+        {
+            List<ReceiptResponse> result = this.smartContractsController.ReceiptSearch(contractAddress, eventName, topics, fromBlock, toBlock);
```
Comment (fassadlr) [inline src/Stratis.Features.Unity3dApi/Controllers/Unity3dController.cs:540]: Hi Ivan,

It seems that the logic in `SmartContractsController.ReceiptSearch` is now shared between 2 controllers. There is already a `smartContractTransactionService` in this controller, so move the logic in `SmartContractsController.ReceiptSearch` there and then just call `smartContractTransactionService.ReceipSearch` from here (also just update `SmartContractsController` to do the same.

😎

Comment (noescape00) [reply src/Stratis.Features.Unity3dApi/Controllers/Unity3dController.cs]: `It seems that the logic in SmartContractsController.ReceiptSearch is now shared between 2 controllers.`
Logic was just in SC controller and unity controller called it (same was done with other methods as well)

But anyway, changes applied

**Review by fassadlr — APPROVED** [review]
I know its a bit more code bit this looks a lot cleaner now ;) I was worried also that some person randomly moves or even remove the sc controller and then your code breaks.

**Comment by noescape00** [comment]
Closing this one and moving it to https://github.com/stratisproject/StratisFullNode/pull/712

**Pull request closed without merging**